app/services/metrics.rb

```diff
@@ -0,0 +1,105 @@
+class Metrics
+  attr_accessor :prison_name, :grouped_by_prison, :visits
+
+  def visit_count
+    result = execute_query_and_type_result <<-SQL
+      SELECT COUNT(*)::integer FROM visits
+    SQL
+    result.values.flatten.first
+  end
+
+  def visit_count_by_state
+    result = execute_query_and_type_result <<-SQL
+      SELECT processing_state,
+             COUNT(*)::integer
+      FROM visits
+      GROUP BY processing_state
+    SQL
+    result.map(&:values).to_h
+  end
+
+  def visit_count_by_prison_and_state
+    result = execute_query_and_type_result <<-SQL
+      SELECT prisons.name AS prison_name,
+             processing_state,
+             COUNT(*)
+        FROM visits
+      INNER JOIN prisons ON prisons.id = visits.prison_id
+      GROUP BY processing_state,
+               prison_name
```

Comment:
I'm wondering whether we should make these database views. We could map them to Active Record objects and it might make the code more natural on the Ruby side.